### PR TITLE
add logs to bootstrap and add cronFetcher script

### PR DESCRIPTION
**Pull request by berthieresteban**
## What does this PR do?

This PR add `cronFetcher.js` script that fetch for each crypto currency referenced in crypto_list table:
  - the last minut every minut and delete the oldest
  - the last hour every hour and delete the oldest
  - the last day every day and delete the oldest

### How should this be manually tested?

You'll need the api key for cryptocompare api (teams).
You can set it in API_KEY env variable or with the `--api_key blablabla` parameter.

  - `cd backend`
  - `npm i`
  - `cd scripts`
  - `./init-db.sh`
  - `node bootstrap.js` (you can see the beautiful logs... :smile:)
  - `node cronFetcher.js`
  - Logs will tells you when new rows are add/removed from tables
  - You can verify the changes from your postgres console (`psql -h localhost -U postgres`)

### Other changes
This PR add logs with progress bars to bootstrap script.

### Boyscout
/

### CODE REVIEW

File: backend/scripts/bootstrap.js [hunk 1]
```diff
@@ -23,15 +24,23 @@ client.connect()
 const cryptos = ['BTC', 'ETH', 'XRP', 'LTC', 'BCH', 'USDT', 'EOS', 'BNB', 'BSV', 'TRX']
 const symbol = 'EUR'
 const api_key = argv.api_key;
+const numberOfDays = 60;
+const numberOfHours = 48;
+const numberOfMinuts = 120;
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: These global should be in uppercase

File: backend/scripts/bootstrap.js [hunk 2]
```diff
@@ -23,15 +24,23 @@ client.connect()
 const cryptos = ['BTC', 'ETH', 'XRP', 'LTC', 'BCH', 'USDT', 'EOS', 'BNB', 'BSV', 'TRX']
 const symbol = 'EUR'
 const api_key = argv.api_key;
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: The global should be in uppercase

File: backend/scripts/bootstrap.js [hunk 3]
```diff
@@ -23,15 +24,23 @@ client.connect()
...
+const numberOfDays = 60;
+const numberOfHours = 48;
+const numberOfMinuts = 120;
 
 // GET ALL CRYPTOS INFOS
 const getCryptoList = async () => {
     const url = `https://min-api.cryptocompare.com/data/all/coinlist`;
     let query = 'INSERT INTO crypto_list (symbol, fullname, picture_url) VALUES\n';
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: Please name properly each query

File: backend/scripts/bootstrap.js [hunk 4]
```diff
@@ -23,15 +24,23 @@ client.connect()
...
+  
+    const cryptoListBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+    cryptoListBar.start(cryptos.length, 0);
+  
     await axios.get(url)
         .then(response => {
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: wtf ? You use `await` and `.then`. That does not make any sense. use async/await, or promises chain, but not both

File: backend/scripts/bootstrap.js [hunk 5]
```diff
@@ -43,17 +52,23 @@ const getCryptoList = async () => {
     await client
         .query(query)
         .catch(e => console.error(e.stack))
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: same remark than above

File: backend/scripts/bootstrap.js [hunk 6]
```diff
@@ -43,17 +52,23 @@ const getCryptoList = async () => {
...
+    cryptoListBar.stop()
 }
 
 // GET EACH CRYPTO VALUES FOR 60 LAST DAYS
 const getCryptosDays = async () => {
     let query = 'INSERT INTO crypto_history (crypto_id, period, timestamp, open, high, low, close) VALUES \n';
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: please name properly your queries

File: backend/scripts/bootstrap.js [hunk 7]
```diff
@@ -43,17 +52,23 @@ const getCryptoList = async () => {
...
+
+    const cryptoDaysBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+    cryptoDaysBar.start(numberOfDays * cryptos.length, 0);
+
     for (const [index, crypto] of cryptos.entries()) {
-        const url = `https://min-api.cryptocompare.com/data/v2/histoday?fsym=${crypto}&tsym=${symbol}&limit=60&api_key=${api_key}`;
+        const url = `https://min-api.cryptocompare.com/data/v2/histoday?fsym=${crypto}&tsym=${symbol}&limit=${numberOfDays}&api_key=${api_key}`;
         await axios.get(url)
             .then(response => {
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: Same. Use async/await OR promises chain.

File: backend/scripts/bootstrap.js [hunk 8]
```diff
@@ -65,18 +80,24 @@ const getCryptosDays = async () => {
     await client
         .query(query)
         .catch(e => console.error(e.stack))
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: Same

File: backend/scripts/bootstrap.js [hunk 9]
```diff
@@ -65,18 +80,24 @@ const getCryptosDays = async () => {
...
     let query = 'INSERT INTO crypto_history (crypto_id, period, timestamp, open, high, low, close) VALUES \n';
+
+    const cryptoHoursBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+    cryptoHoursBar.start(numberOfHours * cryptos.length, 0);
+
     for (const [index, crypto] of cryptos.entries()) {
-        const url = `https://min-api.cryptocompare.com/data/v2/histohour?fsym=${crypto}&tsym=${symbol}&limit=48&api_key=${api_key}`;
+        const url = `https://min-api.cryptocompare.com/data/v2/histohour?fsym=${crypto}&tsym=${symbol}&limit=${numberOfHours}&api_key=${api_key}`;
         await axios.get(url)
             .then(response => {
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: Same

File: backend/scripts/bootstrap.js [hunk 10]
```diff
@@ -89,18 +110,24 @@ const getCryptosHours = async () => {
     await client
         .query(query)
         .catch(e => console.error(e.stack))
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: Same

File: backend/scripts/bootstrap.js [hunk 11]
```diff
@@ -89,18 +110,24 @@ const getCryptosHours = async () => {
...
+
+    const cryptoMinutsBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+    cryptoMinutsBar.start(numberOfMinuts * cryptos.length, 0);
+
     for (const [index, crypto] of cryptos.entries()) {
-        const url = `https://min-api.cryptocompare.com/data/v2/histominute?fsym=${crypto}&tsym=${symbol}&limit=120&api_key=${api_key}`;
+        const url = `https://min-api.cryptocompare.com/data/v2/histominute?fsym=${crypto}&tsym=${symbol}&limit=${numberOfMinuts}&api_key=${api_key}`;
         await axios.get(url)
             .then(response => {
                 const minuts = response.data.Data.Data;
```
Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]: Same (i stop there)

Comment (Yoann-Abbes) [inline backend/scripts/bootstrap.js]:
```suggestion
                const minuts = response.data.Data.Data;
```
nit : should be names `minutes`

File: backend/scripts/cronFetcher.js [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+    host: argv.host,
+    database: argv.database,
+    password: argv.password,
+    port: argv.port
+})
+
+client.connect()
+
+const symbol = 'EUR'
+const api_key = argv.api_key;
```
Comment (Yoann-Abbes) [inline backend/scripts/cronFetcher.js]: These globals should be in uppercase

File: backend/scripts/cronFetcher.js [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+const deleteLastPeriod = async (period) => {
+    const cryptos = await getCryptoListDb();
+    for (const crypto of cryptos) {
+        const query = `DELETE FROM crypto_history WHERE id IN (SELECT id FROM crypto_history WHERE period = '${period}' AND crypto_id = ${crypto.id} ORDER BY timestamp LIMIT 1);`
+        await client
+            .query(query)
+            .then(() => {
+                console.log(`Latest ${period} successfully deleted!`)
+            })
+            .catch(e => console.error(e.stack))
```
Comment (Yoann-Abbes) [inline backend/scripts/cronFetcher.js]: Well you should know what i want to say by now :D

File: backend/scripts/bootstrap.js [hunk 14]
```diff
@@ -16,112 +17,133 @@ const client = new Client({
...
+    const cryptoDaysBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+    cryptoDaysBar.start(NUMBER_OF_DAYS * SELECTED_CRYPTOS.length, 0);
+
+    for (const [index, crypto] of SELECTED_CRYPTOS.entries()) {
+        const cryptoDaysUrl = `https://min-api.cryptocompare.com/data/v2/histoday?fsym=${crypto}&tsym=${DEVISE}&limit=${NUMBER_OF_DAYS}&api_key=${APIKEY}`;
+        const cryptoDaysResponse = await axios.get(cryptoDaysUrl);
+        const days = cryptoDaysResponse.data.Data.Data;
+        for (const [indexDay, day] of days.entries()) {
+            cryptoDaysBar.update(indexDay + index * NUMBER_OF_DAYS);
+            cryptoDaysQuery += `('${index + 1}', 'days', to_timestamp(${day.time}), '${day.open}', '${day.high}', '${day.low}', '${day.close}'),\n`;
```
Comment (Gerroll) [inline backend/scripts/bootstrap.js]:
```suggestion
            cryptoDaysQuery += `('${index + 1}', 'daily', to_timestamp(${day.time}), '${day.open}', '${day.high}', '${day.low}', '${day.close}'),\n`;
```

File: backend/scripts/bootstrap.js [hunk 15]
```diff
@@ -16,112 +17,133 @@ const client = new Client({
...
+    const cryptoHoursBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+    cryptoHoursBar.start(NUMBER_OF_HOURS * SELECTED_CRYPTOS.length, 0);
+
+    for (const [index, crypto] of SELECTED_CRYPTOS.entries()) {
+        const cryptoHoursUrl = `https://min-api.cryptocompare.com/data/v2/histohour?fsym=${crypto}&tsym=${DEVISE}&limit=${NUMBER_OF_HOURS}&api_key=${APIKEY}`;
+        const cryptoHoursResponse = await axios.get(cryptoHoursUrl);
+        const hours = cryptoHoursResponse.data.Data.Data;
+        for (const [indexHour, hour] of hours.entries()) {
+            cryptoHoursBar.update(indexHour + index * NUMBER_OF_HOURS);
+            cryptoHoursQuery += `('${index + 1}', 'hours', to_timestamp(${hour.time}), '${hour.open}', '${hour.high}', '${hour.low}', '${hour.close}'),\n`;
```
Comment (Gerroll) [inline backend/scripts/bootstrap.js]:
```suggestion
            cryptoHoursQuery += `('${index + 1}', 'hourly', to_timestamp(${hour.time}), '${hour.open}', '${hour.high}', '${hour.low}', '${hour.close}'),\n`;
```

File: backend/scripts/bootstrap.js [hunk 16]
```diff
@@ -16,112 +17,133 @@ const client = new Client({
...
+    const cryptoMinutsBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+    cryptoMinutsBar.start(NUMBER_OF_MINUTES * SELECTED_CRYPTOS.length, 0);
+
+    for (const [index, crypto] of SELECTED_CRYPTOS.entries()) {
+        const cryptoMinutsUrl = `https://min-api.cryptocompare.com/data/v2/histominute?fsym=${crypto}&tsym=${DEVISE}&limit=${NUMBER_OF_MINUTES}&api_key=${APIKEY}`;
+        const cryptoMinutsResponse = await axios.get(cryptoMinutsUrl);
+        const minutes = cryptoMinutsResponse.data.Data.Data;
+        for (const [indexMinut, minute] of minutes.entries()) {
+            cryptoMinutsBar.update(indexMinut + index * NUMBER_OF_MINUTES);
+            cryptoMinutsQuery += `('${index + 1}', 'minuts', to_timestamp(${minute.time}), '${minute.open}', '${minute.high}', '${minute.low}', '${minute.close}'),\n`;
```
Comment (Gerroll) [inline backend/scripts/bootstrap.js]:
```suggestion
            cryptoMinutsQuery += `('${index + 1}', 'minute', to_timestamp(${minute.time}), '${minute.open}', '${minute.high}', '${minute.low}', '${minute.close}'),\n`;
```

**Pull request merged**